include/spore/codegen/codegen_app.hpp

```diff
@@ -123,6 +123,18 @@ namespace spore::codegen
                         throw codegen_error(codegen_error_code::rendering, "failed to convert input data to json, file={}", input.string());
                     }
 
+                    if (options.dump_ast)
+                    {
+                        std::filesystem::path output_filename = fmt::format("{}.dump", input.stem().string());
+                        std::filesystem::path output_directory = std::filesystem::path(options.output) / "ast" / input.parent_path();
+                        std::string output = (output_directory / output_filename).string();
+
+                        if (!spore::codegen::write_file(output, json_data.dump(2)))
+                        {
+                            throw codegen_error(codegen_error_code::io, "failed to write output, file={}", output);
```

Comment:
Should it fail if failing to dump the AST? I think we could log a warning in this case and it should be enough, what do you think?